### DISCUSSION TITLE
NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE_WITH_DEFAULT

### DISCUSSION THREAD

**Discussion by ansa-sequence** [nlohmann/json#3653] [Q&A] [answered]
Why we have not save argument order in output json file?

For example : 
[Code example](https://pastebin.com/ULq7tKPN)

After compilation this program we haven not argument order json file
Output : 
`
{
    "windowFlags": 0,
    "windowPos": [
        0,
        0
    ],
    "windowSize": [
        1920,
        1080
    ],
    "windowTitle": "Engine"
}
`

Its standard behavior or we could do it in right order?
Because intuitively we are thinking library has this behavior.

**Comment by falbrechtskirchinger** [comment, marked as answer]
`nlohmann::json` is unordered. You'd have to use `nlohmann::ordered_json` but we don't have a macro for that yet (there's a PR in the works). You can do it manually, though.

Also, why are you using the non-intrusive version of the macro in an intrusive manner?

**Reply by ansa-sequence** [reply]
Thank you for recommendation to use intrusive version of macro, its more comfortable =). I will wait updated versions of this macro!
